lib/Dialect/RTL/RTLFolds.cpp

```diff
@@ -856,6 +859,15 @@ struct ICmpCanonicalizeConstant final : public OpRewritePattern<ICmpOp> {
       };
 
       switch (op.predicate()) {
+      case ICmpPredicate::eq:
+        // eq(x, 0) -> not(orr(x))
+        if (rhs.isNullValue()) {
+          Value orR =
+              rewriter.create<OrROp>(op.getLoc(), op.getType(), op.lhs());
+          rewriter.replaceOpWithNewOp<XorOp>(op, orR, getConstant(APInt(1, 1)));
+          return success();
```

Comment:
I'd recommend that we canonicalize things the other way: "eq" is going to be easier to reason about.

Should we drop "orr" and "andr" from the RTL dialect entirely and just use comparisons?  We could then rename xorr to "parity"